pkg/util/tests/kubectl/kubectl.go

```diff
@@ -0,0 +1,317 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR Condition OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kubectl
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/istio-ecosystem/sail-operator/pkg/util/tests/shell"
+	r "github.com/istio-ecosystem/sail-operator/pkg/util/tests/types"
+)
+
+const DefaultBinary = "kubectl"
+
+var (
+	ErrNotFound       = errors.New("resource was not found")
+	EmptyResourceList = r.ResourceList{
+		APIVersion: "v1",
+		Items:      []interface{}{},
+		Kind:       "List",
+		Metadata:   r.Metadata{ResourceVersion: ""},
+	}
+)
+
+// kubectl return the kubectl command
+// If the environment variable COMMAND is set, it will return the value of COMMAND
+// Otherwise, it will return the default value "kubectl" as default
+// Arguments:
+// - format: format of the command without kubeclt or oc
+// - args: arguments of the command
+func kubectl(format string, args ...interface{}) string {
+	binary := DefaultBinary
+	if cmd := os.Getenv("COMMAND"); cmd != "" {
+		binary = cmd
+	}
+
+	return binary + " " + fmt.Sprintf(format, args...)
+}
+
+// ApplyString applies the given yaml string to the cluster
+func ApplyString(yamlString string) error {
+	cmd := kubectl("apply --server-side -f -")
+	_, err := shell.ExecuteCommandWithInput(cmd, yamlString)
+	if err != nil {
+		return fmt.Errorf("error applying yaml: %v", err)
+	}
+
+	return nil
+}
+
+// DeleteString delete the given yaml string to the cluster
+func DeleteString(yamlString string) error {
+	cmd := kubectl("delete -f -")
+	output, err := shell.ExecuteCommandWithInput(cmd, yamlString)
+	if err != nil {
+		// Workaround because the resource can be already deleted
+		if strings.Contains(output, "not found") {
+			return nil
+		}
+
+		return fmt.Errorf("error deleting yaml: %v", err)
+
+	}
+
+	return nil
+}
+
+// GetConditions returns the condition of a resource
+func GetConditions(ns, resourceType, resourceName string) ([]r.Condition, error) {
```

Comment:
This uses `resourceType`, but one of the other functions uses `kind`. Can we use just `kind` and `name` across all functions (instead of `resourceType` and `resourceName`).